cli/internal/packagemanager/packagemanager.go

```diff
@@ -144,3 +152,34 @@ func (pm PackageManager) GetWorkspaces(rootpath fs.AbsolutePath) ([]string, erro
 func (pm PackageManager) GetWorkspaceIgnores(rootpath fs.AbsolutePath) ([]string, error) {
 	return pm.getWorkspaceIgnores(pm, rootpath)
 }
+
+// GetPackageManagerVersionFromCmd returns the version printed to stdio
+// from running `<pkgExe> --version`. This style works for all supported
+// package managers.
+func GetPackageManagerVersionFromCmd(pm *PackageManager, projectDirectory string) (string, error) {
+	cmd := exec.Command(pm.Command, "--version")
+	cmd.Dir = projectDirectory
+	out, err := cmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("could not detect %s version: %v", pm.Name, err)
+	}
+	return strings.TrimSpace(string(out)), nil
+}
+
+// GetPackageManagerVersionFromCmdPanic returns the version printed to stdio
+// from running `<pkgExe> --version`
+func GetPackageManagerVersionFromCmdPanic(pm *PackageManager, projectDirectory string) string {
```

Comment:
this may not be a desired pattern. if not, please forgive, i don't know `go` conventions well. in `ocaml`, when we have something that may return a `Result` or `Option` type, if we want to call a fn that _must_ return a value or panic, instead of calling `fn_name` we call `fn_name_exn`. it generally calls the original fn, inspects the result type for the Err variant (e.g. 2nd tuple entry), and panics on existence. that's what i've done in this little sugary fn